File: signal-matrix/javascript/signals/dlp_unknown.js
'use strict';
/**
 * UNKNOWN: DLP — module required, only anonymous profile returned.
 * PII taint functions exist but are never called from server.js.
 */
const axios = require('axios');

function getAnonProfile(userId) {
    return { id: userId, role: 'user' }; // safe — no PII
}

function exportPiiUnknown(user) {
    // DLP UNKNOWN: SSN + card sent to external — module required but never called
    axios.post('https://crm.example.com/sync', { ssn: user.ssn, card: user.creditCard });
}

function logPiiUnknown(ssn, dob) {
    // DLP UNKNOWN: PII to console — never called
    console.log(`ssn=${ssn} dob=${dob}`);
}

module.exports = { getAnonProfile, exportPiiUnknown, logPiiUnknown };

File: signal-matrix/python/signals/dlp_unknown.py
"""UNKNOWN: DLP — module imported, safe function called, PII taint function never reached."""

def get_public_profile(user_id: int) -> dict:
    """Safe — returns non-PII data. Called from entrypoint."""
    return {"user_id": user_id, "role": "user", "active": True}

def export_pii_unknown(user_data: dict) -> str:
    """
    DLP UNKNOWN: SSN + credit card flow to external API here.
    Module IS imported (so not NOT_REACHABLE) but this function is never called
    from any traced entrypoint (so not REACHABLE).
    """
    import requests
    ssn  = user_data.get("ssn")
    card = user_data.get("credit_card")
    # DLP taint: PII → external HTTP sink
    requests.post("https://analytics.example.com/track", json={"ssn": ssn, "card": card})
    return f"exported:{ssn}"

def log_pii_unknown(user: dict) -> None:
    """DLP UNKNOWN: PII into logger, never called."""
    import logging
    logging.getLogger(__name__).info(
        f"user ssn={user.get('ssn')} dob={user.get('dob')} card={user.get('credit_card')}"
    )
